src/wp-includes/media.php

```diff
@@ -5564,6 +5686,123 @@ function wp_get_loading_attr_default( $context ) {
 	return 'lazy';
 }
 
+/**
+ * Gets loading optimization attributes.
+ *
+ * @since 6.3.0
+ *
+ * @param string  $tag_name The tag name.
+ * @param array[] $attr     Array of the attributes for the tag.
+ * @param string  $context  Context for the element for which the loading optimization attribute is requested.
+ *
+ * @return array[] Loading optimization attributes.
+ */
+function wp_get_loading_optimization_attributes( $tag_name, $attr, $context ) {
+	global $wp_query;
+
+	$loading_attrs = array();
+
+	// For now this function only supports images.
+	if ( 'img' !== $tag_name ) {
+		return $loading_attrs;
+	}
+
+	// For any resources, width and height must be provided, to avoid layout shifts.
+	if ( ! isset( $attr['width'] ) || ! isset( $attr['height'] ) ) {
+		return $loading_attrs;
+	}
+
+	if ( isset( $attr['loading'] ) && 'lazy' === $attr['loading'] ) {
+		if ( isset( $attr['fetchpriority'] ) && 'high' === $attr['fetchpriority']
+		) {
+			_doing_it_wrong(
+				__FUNCTION__,
+				sprintf(
+					/* translators: %s: fetchpriority="high". */
+					__( 'An image cannot be lazy-loaded and assigned %s at the same time.' ),
+					'<code>fetchpriority="high"</code>'
+				),
+				'6.3.0'
+			);
+		}
+
+		$loading_attrs['loading'] = 'lazy';
+		return wp_maybe_add_fetchpriority_high_attr( $loading_attrs, $attr );
+	}
+
+	// An image with `fetchpriority="high"` cannot be assigned `loading="lazy"` at the same time.
+	if ( isset( $attr['fetchpriority'] ) && 'high' === $attr['fetchpriority'] ) {
+		return wp_maybe_add_fetchpriority_high_attr( $loading_attrs, $attr );
+	}
+
+	/*
+	 * Skip lazy-loading for the overall block template, as it is handled more granularly.
+	 * The skip is also applicable for `fetchpriority`.
+	 */
+	if ( 'template' === $context ) {
+		return $loading_attrs;
+	}
+
+	// Do not lazy-load images in the header block template part, as they are likely above the fold.
+	// For classic themes, this is handled in the condition below using the 'get_header' action.
```

Comment:
```suggestion
	/*
	 * Do not lazy-load images in the header block template part, as they are likely above the fold.
	 * For classic themes, this is handled in the condition below using the 'get_header' action.
	 */
```
Change to multi-line comment format